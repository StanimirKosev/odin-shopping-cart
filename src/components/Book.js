import React from "react";

export const Book = ({ name, image, id, books, handleChange }) => {
  const keys = Object.keys(books);

  return (
    <div className="book-box">
      <img className="book-img" alt={name} src={image} />
      <label>
        Quantity:
        <input
          type="number"
          min="0"
          name={keys[id]} // name attribute corresponds to the state name
          value={books[keys[id]]} // value attr correspond to the state value
          onChange={handleChange}
        />
      </label>
    </div>
  );
};
