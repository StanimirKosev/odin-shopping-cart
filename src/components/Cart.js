import React from "react";
import { useNavigate } from "react-router-dom";
import backgroud from "../images/background-2.avif";

export const Cart = () => {
  let navigate = useNavigate();

  return (
    <div className="cart-page">
      <img alt="backgroud-cart" src={backgroud} />
      <button
        type="submit"
        className="checkout-btn"
        onClick={() => navigate("/")}
      >
        Checkout
      </button>
      <div className="quote-two">
        <div className="quote">
          For books are more than books, they are the life, the very heart and
          core of ages past, the reason why men worked and died, the essence and
          quintessence of their lives.
        </div>
        <div className="author">Marcus Tullius Cicero</div>
      </div>
    </div>
  );
};
