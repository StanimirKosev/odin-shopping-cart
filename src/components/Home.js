import React from "react";
import backgroud1 from "../images/background-1.avif";

export const Home = () => {
  return (
    <div className="home">
      <div className="quote-one">
        <div className="quote">
          Read at every wait; read at all hours; read within leisure; read in
          times of labor; read as one goes in; read as one goes out. The task of
          the educated mind is simply put: read to lead.
        </div>
        <div data-testid="author" className="author">
          Marcus Tullius Cicero
        </div>
      </div>
      <img alt="background" src={backgroud1} />
    </div>
  );
};
