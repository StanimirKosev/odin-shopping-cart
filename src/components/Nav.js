import React from "react";
import { Link } from "react-router-dom";
import bag from "../images/bag.png";

export const Nav = ({ cartItems }) => {
  return (
    <nav>
      <Link to="/odin-shopping-cart">
        <h1>Book e-commerce</h1>
      </Link>
      <div className="links">
        <Link to="/odin-shopping-cart">Home</Link>
        <Link to="/odin-shopping-cart/shop">Shop</Link>
        <Link to="/odin-shopping-cart/cart" className="link-cart">
          <div className="cart-icon">
            <div className="cart-quantity">{cartItems}</div>
            <img alt="bag-img" src={bag} />
          </div>
        </Link>
      </div>
    </nav>
  );
};
