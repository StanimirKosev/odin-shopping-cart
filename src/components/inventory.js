import rulesForLife from "../images/12-rules-for-life.jpg";
import beyondOrder from "../images/beyond-order.jpg";
import crimeAndPunishment from "../images/crime-and-punishment.jpg";
import extremeOwnership from "../images/extreme-ownership.jpg";
import theChangingWorldOrder from "../images/ray-dalio.jpg";
import almanackOfNavalRavikant from "../images/the-almanack-of-naval-ravikant.jpg";
import theBitcoinStandart from "../images/the-bitcoin-standart.jpg";
import thePsychologyofMoney from "../images/the-psychology-of-money.jpg";
import whyWeSleep from "../images/why-we-sleep.jpg";
import deepWork from "../images/deep-work.jpg";

export const inventory = [
  {
    name: "The Almanack of Naval Ravikant: A Guide to Wealth and Happiness",
    image: almanackOfNavalRavikant,
    id: 0,
  },
  {
    name: "The Bitcoin Standard: The Decentralized Alternative to Central Banking",
    image: theBitcoinStandart,
    id: 1,
  },
  {
    name: "12 Rules for Life: An Antidote to Chaos",
    image: rulesForLife,
    id: 2,
  },
  {
    name: "Beyond Order: 12 More Rules For Life",
    image: beyondOrder,
    id: 3,
  },
  {
    name: "The Changing World Order: Why Nations Succeed and Fail",
    image: theChangingWorldOrder,
    id: 4,
  },
  {
    name: "The Psychology of Money",
    image: thePsychologyofMoney,
    id: 5,
  },
  {
    name: "Extreme Ownership: How U.S. Navy SEALs Lead and Win",
    image: extremeOwnership,
    id: 6,
  },
  {
    name: "Deep Work: Rules for Focused Success in a Distracted World",
    image: deepWork,
    id: 7,
  },
  {
    name: "Why We Sleep: Unlocking the Power of Sleep and Dreams",
    image: whyWeSleep,
    id: 8,
  },
  {
    name: "Crime and Punishment",
    image: crimeAndPunishment,
    id: 9,
  },
];
