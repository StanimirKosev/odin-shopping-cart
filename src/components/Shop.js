import React from "react";
import { useNavigate } from "react-router-dom";
import { inventory } from "./inventory";
import { Book } from "./Book";

export const Shop = ({ books, handleChange }) => {
  let navigate = useNavigate();

  return (
    <div className="shop-page">
      <button
        onClick={() => navigate("/odin-shopping-cart/cart")}
        type="button"
        className="shop-button"
      >
        Cart
      </button>
      {inventory.map((book) => (
        <Book
          key={book.id}
          name={book.name}
          image={book.image}
          id={book.id}
          books={books}
          handleChange={handleChange}
        />
      ))}
    </div>
  );
};
