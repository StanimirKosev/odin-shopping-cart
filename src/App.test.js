/* eslint-disable jest/valid-expect */
import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Nav } from "./components/Nav";
import { Home } from "./components/Home";

// practicing testing
// rendering elements and taking a snapshot
describe("Nav", () => {
  it("renders heading", () => {
    render(<Nav />, { wrapper: MemoryRouter });
    expect(screen.getByRole("heading", { name: /book e-commerce/i }));
  });

  it("renders Nav component", () => {
    const { container } = render(<Nav />, { wrapper: MemoryRouter });
    expect(container).toMatchSnapshot();
  });
});

describe("Home tests", () => {
  it("renders quote author", () => {
    render(<Home />);
    expect(screen.getByTestId("author", { name: /marcus tullius cicero/i }));
  });

  it("renders Home component", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
