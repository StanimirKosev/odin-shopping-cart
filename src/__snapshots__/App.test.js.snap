// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests renders Home component 1`] = `
<div>
  <div
    class="home"
  >
    <div
      class="quote-one"
    >
      <div
        class="quote"
      >
        Read at every wait; read at all hours; read within leisure; read in times of labor; read as one goes in; read as one goes out. The task of the educated mind is simply put: read to lead.
      </div>
      <div
        class="author"
        data-testid="author"
      >
        Marcus Tullius Cicero
      </div>
    </div>
    <img
      alt="background"
      src="background-1.avif"
    />
  </div>
</div>
`;

exports[`Nav renders Nav component 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      <h1>
        Book e-commerce
      </h1>
    </a>
    <div
      class="links"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/shop"
      >
        Shop
      </a>
      <a
        class="link-cart"
        href="/cart"
      >
        <div
          class="cart-icon"
        >
          <div
            class="cart-quantity"
          />
          <img
            alt="bag-img"
            src="bag.png"
          />
        </div>
      </a>
    </div>
  </nav>
</div>
`;
